src/org/zaproxy/zap/extension/ascanrulesAlpha/IntegerOverflow.java

```diff
@@ -0,0 +1,271 @@
+ * limitations under the License.
+ */
+
+/**
+ * Integer Overflow an active scan rule
+ * Copyright (C) 2015 Institute for Defense Analyses
+ * @author Mark Rader based upon the example active scanner by psiinon
+ * 
+ */
+
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+
+
+
+
+import org.apache.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractAppParamPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpResponseHeader;
+import org.parosproxy.paros.network.HttpStatusCode;
+
+
+public class IntegerOverflow extends AbstractAppParamPlugin  {
+
+	/**
+	 * Prefix for internationalised messages used by this rule
+	 */
+	private static final String MESSAGE_PREFIX = "ascanalpha.integeroverflow.";
+	private static final int PLUGIN_ID = 30003;
+	private static Logger log = Logger.getLogger(IntegerOverflow.class);
+	
+	@Override
+	public int getId() {
+		return PLUGIN_ID;
+	}
+
+	@Override
+	public String getName() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "name");
+	}
+
+	@Override
+	public String[] getDependency() {
+		return null;
+	}
+
+	@Override
+	public String getDescription() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "desc");
+	}
+
+	@Override
+	public int getCategory() {
+		return Category.INJECTION;
+	}
+
+	@Override
+	public String getSolution() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "soln");
+	}
+
+	@Override
+	public String getReference() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "refs");
+	}
+	
+	public String getOther() {
+		return Constant.messages.getString(MESSAGE_PREFIX + "other");
+	}
+	
+	private String getError(char c) {
+		return Constant.messages.getString(MESSAGE_PREFIX + "error"+c);
+	}
+	
+	@Override
+	public void init() {
+
+	}
+
+	/*
+	 * This method is called by the active scanner for each GET and POST parameter for every page 
+	 * @see org.parosproxy.paros.core.scanner.AbstractAppParamPlugin#scan(org.parosproxy.paros.network.HttpMessage, java.lang.String, java.lang.String)
+	 */
+	@Override
+	public void scan(HttpMessage msg, String param, String value) {
+	
+		if (this.isStop()) { // Check if the user stopped things
+			if (log.isDebugEnabled()) {
+				log.debug("Scanner "+this.getName()+" Stopping.");
+			}
+			return; // Stop!
+		}
+		
+		try {
+			
+			msg = getNewMsg();
+			String returnAttack = randomIntegerString(4);// The number of full length ints to send.
+			setParameter(msg, param, returnAttack);
+			sendAndReceive(msg);
+			HttpResponseHeader requestReturn = msg.getResponseHeader();			
+			// This is where BASE baseResponseBody was you detect potential vulnerabilities in the response
+    		String chkerrorheader = requestReturn.getHeadersAsString();
+    		log.debug("Header: "+ chkerrorheader);

```

Comment:
Might worth doing this only if debug is enabled (the same comment for the other log of `chkerrorheader`).